tests/integration/targets/aws_glue_crawler/tasks/main.yml

```diff
@@ -0,0 +1,333 @@
+---
+- name: aws_glue_crawler integration tests
+  collections:
+    - amazon.aws
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+    # AWS CLI is needed until there's a module to get info about Glue crawlers
+    - name: Install AWS CLI
+      pip:
+        name: awscli
+        state: present
+
+    - name: Create minimal Glue crawler role
+      iam_role:
+        name: "{{ glue_crawler_role_name }}"
+        assume_role_policy_document:
```

Comment:
Something seems incorrect about this when I test with the policy PR, I get this failure:
```
    "message": "Service is unable to assume role arn:aws:iam::966509639900:role/ansible-test--glue-crawler. Please verify role's TrustPolicy",
    "msg": "An error occurred (InvalidInputException) when calling the CreateCrawler operation: Service is unable to assume role arn:aws:iam::966509639900:role/ansible-test--glue-crawler. Please verify role's TrustPolicy",
```
I'm not able to test past this point.

Reply:
@jillr That's curious.

The role name in the error message appears to be missing unique ID defined in `tests/integration/targets/aws_glue_crawler/defaults/main.yml`:

```
unique_id: "{{ resource_prefix | regex_search('(\\d+-?)(\\d+)$') }}"
glue_crawler_role_name: "ansible-test-{{ unique_id }}-glue-crawler"
```

Unique ID does show up in the [most recent Zuul log](https://dashboard.zuul.ansible.com/t/ansible/build/70c37601eeb04b35939864d6c1b709e3/console):

```
changed: [testhost] => {
    "arn": "arn:aws:iam::966509639900:role/ansible-test-10-105-glue-crawler",
    "assume_role_policy_document": {
        "statement": [
            {
                "action": "sts:AssumeRole",
                "effect": "Allow",
                "principal": {
                    "service": "glue.amazonaws.com"
                }
            }
        ],
        "version": "2008-10-17"
    },
    "attached_policies": [
        {
            "policy_arn": "arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess",
            "policy_name": "AWSXrayWriteOnlyAccess"
        }
    ],
    "changed": true,
    "create_date": "2021-04-25T19:39:15+00:00",
    "gid": 0,
    "group": "root",
    "iam_role": {
        "arn": "arn:aws:iam::966509639900:role/ansible-test-10-105-glue-crawler",
        "assume_role_policy_document": {
            "statement": [
                {
                    "action": "sts:AssumeRole",
                    "effect": "Allow",
                    "principal": {
                        "service": "glue.amazonaws.com"
                    }
                }
            ],
            "version": "2008-10-17"
        },
        "attached_policies": [
            {
                "policy_arn": "arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess",
                "policy_name": "AWSXrayWriteOnlyAccess"
            }
        ],
        "create_date": "2021-04-25T19:39:15+00:00",
        "max_session_duration": 3600,
        "path": "/",
        "role_id": "AROA6CCDWXDODL4HTQTFF",
        "role_last_used": {},
        "role_name": "ansible-test-10-105-glue-crawler",
        "tags": {}
    },
    "invocation": {
        "module_args": {
            "assume_role_policy_document": "{\"Version\": \"2008-10-17\", \"Statement\": [{\"Action\": \"sts:AssumeRole\", \"Effect\": \"Allow\", \"Principal\": {\"Service\": \"glue.amazonaws.com\"}}]}",
            "aws_access_key": "ASIA6CCDWXDODDIEQYLQ",
            "aws_ca_bundle": null,
            "aws_config": null,
            "aws_secret_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "boundary": null,
            "create_instance_profile": false,
            "debug_botocore_endpoint_logs": true,
            "delete_instance_profile": false,
            "description": null,
            "ec2_url": null,
            "managed_policies": [
                "arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess"
            ],
            "max_session_duration": null,
            "name": "ansible-test-10-105-glue-crawler",
            "path": "/",
            "profile": null,
            "purge_policies": null,
            "purge_tags": true,
            "region": "us-east-1",
            "security_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "tags": null,
            "validate_certs": true
        }
    },
    "max_session_duration": 3600,
    "mode": "0555",
    "owner": "root",
    "path": "/",
    "resource_actions": [
        "iam:CreateRole",
        "iam:ListRoleTags",
        "iam:AttachRolePolicy",
        "iam:ListAttachedRolePolicies",
        "iam:GetRole"
    ],
    "role_id": "AROA6CCDWXDODL4HTQTFF",
    "role_last_used": {},
    "role_name": "ansible-test-10-105-glue-crawler",
    "secontext": "system_u:object_r:root_t:s0",
    "size": 224,
    "state": "directory",
    "tags": {},
    "uid": 0
}
```

I'm fairly confident that the trust policy is correct - I use it for the test crawler's role that I created in my AWS account.

Is there something else that we are missing?